### PR TITLE
New Readme for 0.50

### PR DESCRIPTION
**Pull request by probins**
OK, here's my first stab at this, as discussed in #491. Comments of course welcome. I've removed all the links to other docs until these are (re)written, and all the usage examples until these actually work :-)

Not sure whether you want to keep the existing copyright; neither Luke nor Addy has contributed anything for years. 'Copyright contributors' perhaps?

It might be a good idea to mark 1.0 as deprecated - perhaps rename it too? - and change the master Readme to point to this branch rather than 1.0.

### CODE REVIEW

**Comment by probins** [comment]
just added a further polyfill needed, for `Symbol.iterator`

**Comment by probins** [comment]
might be a good idea to disable travis on this branch for the moment

File: README.md [hunk 1]
```diff
@@ -1,172 +1,54 @@
...
+
+The final specification, though, only defines the [language-specific semantics of modules](http://www.ecma-international.org/ecma-262/6.0/#sec-modules), leaving implementations of the loading of those modules to separate specifications. The dynamic configurable loader is being specified at the [WhatWG loader spec](https://whatwg.github.io/loader/), and from version 0.50 this polyfill implements that.
+
+Support for static loading of modules from HTML using `<script type="module">` tags has also been added to the [WhatWG HTML specification](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element). This adapts the existing script-loading mechanism to include the loading of 'module scripts' together with all the other modules on which those module scripts depend. Unlike the WhatWG loader, however, this is not configurable, and has no connection with the loading pipeline defined there and implemented in this polyfill.
```
Comment (guybedford) [inline README.md]: > Unlike the WhatWG loader, however, this is not configurable, and has no connection with the loading pipeline defined there and implemented in this polyfill.

This isn't quite right. The point is for the WhatWG loader to share the same module registry as the `<script type="module">` tag. And also for the WhatWG loader resolution hooks to affect the loading of the module tag as well. Perhaps @caridy can help confirm this.

File: README.md [hunk 2]
```diff
@@ -1,172 +1,54 @@
...
-
-Running the application:
-```
-> node index.js
-NodeJS test
-```
+Dynamically loads ES6 modules in browsers and NodeJS with support for loading existing and custom module formats through loader hooks.
+
+### Specifications
+ES6/ES2015 introduces native support for Javascript modules, self-standing pieces of code which execute in their own scope and import values exported by other modules. Originally, the specification included the loader, and this polyfill was created to support those loader functions in current browsers and in NodeJS.
```
Comment (guybedford) [inline README.md]: JavaScript not Javascript.

File: README.md [hunk 3]
```diff
@@ -1,172 +1,54 @@
...
+
+Support for static loading of modules from HTML using `<script type="module">` tags has also been added to the [WhatWG HTML specification](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element). This adapts the existing script-loading mechanism to include the loading of 'module scripts' together with all the other modules on which those module scripts depend. Unlike the WhatWG loader, however, this is not configurable, and has no connection with the loading pipeline defined there and implemented in this polyfill.
+
+### Support status June 2016
+* ES6 is now [widely supported](http://kangax.github.io/compat-table/es6/) in the main browsers and in NodeJS.
+* The main browsers/Javascript engines have also implemented the parsing of modules as defined in ES2015; this however is not useful on its own without a loading mechanism, and is not exposed in the browser.
```
Comment (guybedford) [inline README.md]: Again, JavaScript

Comment (guybedford) [inline README.md]: I'm not sure it's worth mentioning implementation status of module syntax at all actually - as you say it is an irrelevant implementation detail of the parser without a loading mechanism.

File: README.md (hunk 1 above)
Comment (guybedford) [inline README.md]: Where have you seen the term `module scripts` used? _Module tag_ is preferable I think.

File: README.md [hunk 4]
```diff
@@ -1,172 +1,54 @@
...
+* Browser support for module scripts is also being implemented, currently behind flags; it's likely to be 2017 before this is usable in production systems.
```
Comment (guybedford) [inline README.md]: I wouldn't estimate a timeline here. We already have the module tag coming in Edge.

File: README.md [hunk 5]
```diff
@@ -1,172 +1,54 @@
...
+* Defines a `Loader` class in the (new to ES2015) `Reflect` global object, with `import`, `resolve` and `load` prototype methods.
+* The `Loader` class also includes a module `Registry`, which is a `Symbol.iterator`.
+* Introduces a `System` global object, with an instance of `Reflect.Loader` as the `loader` property; this is the default browser loader. In addition to the `Loader` methods, this has `resolve`, `fetch`, `translate` and `instantiate` prototype methods, of which the last two are by default no-ops.
+* Defines a `Module` constructor in the `Reflect` global for dynamic construction of modules from within script or module code.
+
+### This polyfill
+* Implements `Reflect.Loader`, providing the ability for developers to create custom loader instances.
+* Implements the module registry
+* Implements `Reflect.Module`
+* Implements `System.loader`, which means that `System.loader.import` can be used to dynamically load ES6 modules.
```
Comment (guybedford) [inline README.md]: Implements the `System.loader` default `System.Loader` class instance?

File: README.md [hunk 6]
```diff
@@ -1,172 +1,54 @@
...
+* Until the parsing of `import` and `export` statements are natively supported, supports both [Traceur](https://github.com/google/traceur-compiler) and [Babel](http://babeljs.io/) for transpiling these statements; this requires loading a large parser, so is not suitable for production use in the browser, though can of course be used in NodeJS without problem.
```
Comment (guybedford) [inline README.md]: Even in NodeJS this is unadvisable in production.

File: README.md [hunk 7]
```diff
@@ -1,172 +1,54 @@
...
+* Fully supports ES6 circular references and live bindings.
```
Comment (guybedford) [inline README.md]: May be worth including a reference link here.

File: README.md [hunk 8]
```diff
@@ -1,172 +1,54 @@
...
+* As only `import` and `export` are transpiled, those wanting to support older browsers should also pre-transpile any ES6 code into ES5.
+* What was previously in the `System` object, for example `System.import`, is now in `System.loader` (applications can of course supply an alias to ease the transition).
+* The previous `normalize` and `locate` methods have now been combined into the `resolve` method.
+* The `Module` constructor has been moved to the `Reflect` object.
+* Although the WhatWG loader will eventually define a `site` object, which will probably include some kind of module name/URL mapping, this is not currently defined and so is not included in the current polyfill; this means there is no longer any `baseURL` or `paths` support. For compatibility with script modules, the default loader only supports relative URLs `/`, `./` and `../` in `import` statements (this is relative to the URL of the importing module).
```
Comment (guybedford) [inline README.md]: An important point here though is that this is a hookable loader - users can make their own loaders with it. That is perhaps the strongest use case for using this loader polyfill for more than experimentation - to be able to have a loader in use that is based on a standard.

File: README.md [hunk 9]
```diff
@@ -1,172 +1,54 @@
...
+
+As before, [SystemJs](https://github.com/systemjs/systemjs/) provides add-on functionality supporting legacy module formats, `System.register`, and module name/URL mapping in the default `System.loader`.
```
Comment (guybedford) [inline README.md]: Perhaps something like - 

For an example of approaches for using the loader in production, see SystemJS which hooks in functionality such as supporting ES6 modules through the `System.register` transpiled format and its own custom module resolution mechanism.

File: README.md [hunk 10]
```diff
@@ -1,172 +1,54 @@
...
 
 ## Contributing
+Changing to the new loader spec is quite a lot of work, so contributions of all kinds are welcome.
```
Comment (guybedford) [inline README.md]: Bold!

File: README.md [hunk 11]
```diff
@@ -1,172 +1,54 @@
...
-Note: you will need to export your username and key to launch it.
-
-  ```sh
-  export SAUCE_USERNAME={your user name} && export SAUCE_ACCESS_KEY={the access key that you see once logged in}
-  npm run test:browsers -- --saucelabs
-  ```
-
 ## Credit
-Copyright (c) 2015 Luke Hoban, Addy Osmani, Guy Bedford
+Copyright (c) 2016 Luke Hoban, Addy Osmani, Guy Bedford
```
Comment (guybedford) [inline README.md]: Let's perhaps update this to:

##Contributors

Peter Robins
Joel Denning
Guy Bedford
Addy Osmani
Luke Hoban

File: README.md (hunk 5 above)
Comment (guybedford) [inline README.md]: `Reflect.Loader` class I mean here.

File: README.md (hunk 1 above)
Comment (probins) [inline README.md]: the HTML spec I link to refers to 'classic scripts' and 'module scripts'

**Comment by guybedford** [comment]
👍

**Pull request merged**